File: EncryptandDecrypt/app/src/main/java/com/ashucode/encryptanddecrypt/MainActivity.java
package com.ashucode.encryptanddecrypt;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.PublicKey;
import java.security.Signature;

import javax.crypto.Cipher;

public class MainActivity extends AppCompatActivity {

    EditText normaltext,encrpytedtext;
    TextView encryptedcode, result;
    Button Encrypt,copy,Decrypt,clear;
    ClipboardManager cpb;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // create a clipboard manager variable to copy text
        cpb = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);

        //Encryption material
        normaltext = findViewById(R.id.normaltext);
        encryptedcode = findViewById(R.id.encryptedcode);
        Encrypt = findViewById(R.id.btn_encrypt);
        copy = findViewById(R.id.btn_copy);

        //Decryption material
        encrpytedtext = findViewById(R.id.dcrypt_etText);
        result = findViewById(R.id.tv_result);
        Decrypt = findViewById(R.id.btn_Dcrypt);
        clear = findViewById(R.id.btn_clear);

        Encrypt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try {
                    //Creating a Signature object
                    Signature sign = Signature.getInstance("SHA256withRSA");

                    //Creating KeyPair generator object
                    KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA");

                    //Initializing the key pair generator
                    keyPairGen.initialize(2048);

                    //Generating the pair of keys
                    KeyPair pair = keyPairGen.generateKeyPair();

                    //Creating a Cipher object
                    Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");

                    //Initializing a Cipher object
                    cipher.init(Cipher.ENCRYPT_MODE, pair.getPublic());

                    //Adding data to the cipher
                    String text = normaltext.getText().toString();
                    byte[] input = text.getBytes();
                    cipher.update(input);

                    //encrypting the data
                    byte[] cipherText = cipher.doFinal();
                    encryptedcode.setText(new String(cipherText, "UTF8"));
//                    System.out.println(new String(cipherText, "UTF8"));
                }
                catch (Exception e)
                {
                    System.out.print(e);
                }
            }
        });

        copy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String data = encryptedcode.getText().toString().trim();

                // check if the textview is not empty
                if (!data.isEmpty()) {

                    // copy the text in the clip board
                    ClipData temp = ClipData.newPlainText("text", data);
                    cpb.setPrimaryClip(temp);

                    // display message that the text has been copied
                    Toast.makeText(MainActivity.this, "Copied", Toast.LENGTH_SHORT).show();
                }
            }
        });

        Decrypt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try {
                    //Creating a Signature object
                    Signature sign = Signature.getInstance("SHA256withRSA");

                    //Creating KeyPair generator object
                    KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA");

                    //Initializing the key pair generator
                    keyPairGen.initialize(2048);

                    //Generate the pair of keys
                    KeyPair pair = keyPairGen.generateKeyPair();

                    //Getting the public key from the key pair
                    PublicKey publicKey = pair.getPublic();

                    //Creating a Cipher object
                    Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");

                    //Initializing a Cipher object
                    cipher.init(Cipher.ENCRYPT_MODE, publicKey);

                    //Add data to the cipher
                    String text2 = normaltext.getText().toString();
                    byte[] input = text2.getBytes();
                    cipher.update(input);

                    //encrypting the data
                    byte[] cipherText = cipher.doFinal();

                    //Initializing the same cipher for decryption
                    cipher.init(Cipher.DECRYPT_MODE, pair.getPrivate());

                    //Decrypting the text
                    byte[] decipheredText = cipher.doFinal(cipherText);
                    result.setText(new String(decipheredText));
                }
                catch (Exception ex)
                {
                    System.out.print(ex);
                }

            }
        });

        clear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String text = encrpytedtext.getText().toString();
                if(text.isEmpty())
                    Toast.makeText(MainActivity.this, "Already Empty !!!", Toast.LENGTH_SHORT).show();
                else
                    encrpytedtext.setText("");
            }
        });

    }
}

File: EncryptionInKotlin/app/src/main/java/com/ashucode/encryptioninkotlin/MainActivity.kt
package com.ashucode.encryptioninkotlin

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.security.KeyPairGenerator
import java.security.Signature
import javax.crypto.Cipher

class MainActivity : AppCompatActivity() {
    lateinit var normalTxt : EditText
    lateinit var encryptText : EditText
    lateinit var encryptCode : TextView
    lateinit var decryptText : TextView
    lateinit var encryptBtn : Button
    lateinit var copy : Button
    lateinit var decryptBtn : Button
    lateinit var clear : Button
    lateinit var cbm : ClipboardManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        cbm = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager

        normalTxt = findViewById(R.id.normaltext)
        encryptCode = findViewById(R.id.encryptedcode)
        encryptText = findViewById(R.id.encryptText)
        decryptText = findViewById(R.id.decryptText)

        encryptBtn = findViewById(R.id.btn_encrypt)
        copy = findViewById(R.id.btn_copy)
        decryptBtn = findViewById(R.id.btn_Dcrypt)
        clear = findViewById(R.id.btn_clear)

        encryptBtn.setOnClickListener()
        {
            try {
                val sign = Signature.getInstance("SHA256withRSA")
                val keypairGen = KeyPairGenerator.getInstance("RSA")
                keypairGen.initialize(2048)
                val pair = keypairGen.generateKeyPair()
                val cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding")
                cipher.init(Cipher.ENCRYPT_MODE, pair.public)
                val text = normalTxt.text.toString()
                val input = text.toByteArray()
                cipher.update(input)
                val cipherText = cipher.doFinal()
                encryptCode.setText(String(cipherText))
            }
            catch (e:Exception)
            {
                System.out.print(e);
            }
        }
        copy.setOnClickListener()
        {
            val data = encryptCode.text.toString().trim()
            if (!data.isEmpty())
            {
                val temp = ClipData.newPlainText("text",data)
                cbm.setPrimaryClip(temp)
                Toast.makeText(this@MainActivity, "Copied", Toast.LENGTH_SHORT).show()
            }
        }
        decryptBtn.setOnClickListener()
        {
            try
            {
                val sign = Signature.getInstance("SHA256withRSA")
                val keyPairGen = KeyPairGenerator.getInstance("RSA")
                keyPairGen.initialize(2048)
                val pair = keyPairGen.generateKeyPair()
                val publicKey = pair.public
                val cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding")
                cipher.init(Cipher.ENCRYPT_MODE, publicKey)
                val text2: String = normalTxt.getText().toString()
                val input = text2.toByteArray()
                cipher.update(input)
                val cipherText = cipher.doFinal()
                cipher.init(Cipher.DECRYPT_MODE, pair.private)
                val decipheredText = cipher.doFinal(cipherText)
                decryptText.setText(String(decipheredText))
            }
            catch (ex: java.lang.Exception)
            {
                print(ex)
            }
        }
        clear.setOnClickListener()
        {
            val txt = encryptText.text.toString()
            if(txt.isEmpty())
                Toast.makeText(this@MainActivity, "Already Empty !!!", Toast.LENGTH_SHORT).show()
            else
            {
                encryptText.setText("")
                decryptText.setText("")
            }

        }
    }
}